File: backend/models/TipoInmueble.js
const mongoose = require('mongoose');

const tipoInmuebleSchema = new mongoose.Schema({
  nombre: { type: String, required: true, unique: true },
  descripcion: { type: String, default: '' },
  activo: { type: Boolean, default: true },
  creado_en: { type: Date, default: Date.now },
  actualizado_en: { type: Date, default: Date.now },
});

tipoInmuebleSchema.pre('save', function (next) {
  this.actualizado_en = Date.now();
  next();
});

module.exports = mongoose.model('TipoInmueble', tipoInmuebleSchema);

File: mobile/lib/models/tipo_inmueble.dart
class TipoInmueble {
  final String id;
  final String nombre;
  final String descripcion;
  final bool activo;

  TipoInmueble({
    required this.id,
    required this.nombre,
    this.descripcion = '',
    this.activo = true,
  });

  Map<String, dynamic> toMap() => {
        'id': id,
        'nombre': nombre,
        'descripcion': descripcion,
        'activo': activo ? 1 : 0,
      };

  factory TipoInmueble.fromMap(Map<String, dynamic> map) => TipoInmueble(
        id: map['id'] as String,
        nombre: map['nombre'] as String,
        descripcion: map['descripcion'] as String? ?? '',
        activo: (map['activo'] as int? ?? 1) == 1,
      );

  factory TipoInmueble.fromJson(Map<String, dynamic> json) => TipoInmueble(
        id: json['_id'] as String,
        nombre: json['nombre'] as String,
        descripcion: json['descripcion'] as String? ?? '',
        activo: json['activo'] as bool? ?? true,
      );
}
